database-plugins/src/main/java/io/cdap/plugin/DBRecord.java

```diff
@@ -124,23 +137,39 @@ public void readFields(ResultSet resultSet) throws SQLException {
       int sqlType = metadata.getColumnType(i + 1);
       int sqlPrecision = metadata.getPrecision(i + 1);
       int sqlScale = metadata.getScale(i + 1);
+      // handles backwards compatibility
+      boolean handleDecimalAsDouble = false;
+      if (sqlType == Types.DECIMAL && outputSchema != null) {
+        Schema outputFieldSchema = field.getSchema();
+        if (outputFieldSchema != null) {
+          outputFieldSchema = outputFieldSchema.isNullable() ? outputFieldSchema.getNonNullable() : outputFieldSchema;
+          handleDecimalAsDouble = outputFieldSchema.getType() == Schema.Type.DOUBLE;
```

Comment:
This does not cover all cases. The schema value changes depending on precision and scale values:
```
type = scale != 0 ? Schema.Type.DOUBLE :
          // with 10 digits we can represent 2^32 and LONG is required
          precision > 9 ? Schema.Type.LONG : Schema.Type.INT;

Reply:
@CuriousVini fixed